quinn-h3/examples/h3.rs

```diff
@@ -486,33 +488,46 @@ impl Future for RecvRequest {
     type Error = Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        let new_state = match &mut self.state {
-            RecvRequestState::Receiving(frames, _) => {
-                let frame = try_ready!(frames.poll());
-                match frame {
-                    None => return Err(format_err!("recieved an empty request")),
-                    Some(HttpFrame::Headers(f)) => {
-                        self.headers = Some(f);
-                        RecvRequestState::Ready
+        loop {
+            match self.state {
+                RecvRequestState::Receiving(ref mut frames, _) => {
+                    println!("recving");
+                    let frame = try_ready!(frames.poll());
+                    match frame {
+                        None => return Err(format_err!("recieved an empty request")),
+                        Some(HttpFrame::Headers(f)) => {
+                            match mem::replace(&mut self.state, RecvRequestState::Decoding(f)) {
+                                RecvRequestState::Receiving(f, s) => self.streams = Some((f, s)),
+                                _ => unreachable!("Invalid state"),
+                            }
+                        }
+                        Some(_) => return Err(format_err!("first frame is not headers")),
```

Comment:
I think we should have `{ }` block around return statements.

Reply:
Is that for readability ?

Reply:
Hmm, maybe it's not actually needed. To me return always seems like more of a statement, and so it's required to be a in a block. But of course it's actually an expression that returns the never type, so maybe this is just me being stuck in my old ways.